lib/expectedConditions.ts

```diff
@@ -301,11 +301,10 @@ export class ProtractorExpectedConditions {
    * @returns {!function} An expected condition that returns a promise
    *     representing whether the URL contains the string.
    */
-  urlContains(url: string): Function {
-    return () => {
-      return this.browser.driver.getCurrentUrl().then((actualUrl: string): boolean => {
-        return actualUrl.indexOf(url) > -1;
-      });
+  urlContains(url: string): (() => Promise<boolean>) {
```

Comment:
This might be better either reverted or changed in a separate commit. I'm not too worried about these methods.

Reply:
Made all changes to EC in this commit.